overlord/snapstate/target.go

```diff
@@ -1253,7 +1257,7 @@ func targetForPathSnap(update PathSnap, snapst SnapState, opts Options) (target,
 		return target{}, fmt.Errorf("cannot install local snap %q: %v != %v (revision mismatch)", update.InstanceName, update.RevOpts.Revision, si.Revision)
 	}
 
-	if update.RevOpts.Channel != "" && update.SideInfo.Channel != "" && update.RevOpts.Channel != update.SideInfo.Channel {
+	if update.RevOpts.Channel != "" && si.Channel != "" && update.RevOpts.Channel != si.Channel {
```

Comment:
what is this change update? is there a matching testt?

Reply:
I think this is from me rebasing things, but `si == update.SideInfo`, it is declared up above.